tests/e2e/framework/testInfo.go

```diff
@@ -184,7 +184,14 @@ func (t testInfo) FetchAndSaveClusterLogs(namespace string) error {
 			}
 			// append logs to file
 			for _, logEntry := range entries {
-				_, err = f.WriteString(fmt.Sprintf("%v\n", logEntry.Payload))
+				timestamp := logEntry.GetTimestamp()
+				fmtTime := time.Unix(timestamp.Seconds, 0)
+				log := fmt.Sprintf("[%02d:%02d:%02d] %s",
+					fmtTime.Hour(), fmtTime.Minute(), fmtTime.Second(), logEntry.GetTextPayload())
+				if log[len(log)-1:] != "\n" {
+					log += "\n"
```

Comment:
is it really the case that sometime there is already a \n and some times there isn't ? it may be quite expensive and maybe should be split
into  (assuming f is buffered and not taking a lock each time)
```
f.WriteString(fmt.Sprintf("[%02d:%02d:%02d] ",fmtTime.Hour(), fmtTime.Minute(), fmtTime.Second())
txt := logEntry.GetTextPayload()
f.WriteString(txt))
if len(txt) == 0 || log[len(log)-1] != '\n' {
  f.WriteByte('\n')
}
```